solutions/usaco-1159.mdx

```diff
@@ -139,4 +139,96 @@ public class ConnectingTwoBarns {
 
 </JavaSection>
 
+</LanguageSection>
+
+## Alternative Implementation - DFS + Binary Search
+						
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+#define forn(i, n) for(ll i = 0; i < n; i++)
+#define pb push_back
+
+const ll MAXN = 1e5;
+
+vl adj[MAXN], comps[MAXN];
+ll comp[MAXN];
+
+void dfs(ll cur, ll c){
+	if(comp[cur] != -1){
+		return;
+	}
+	comp[cur] = c;
+	for(ll u: adj[cur]){
+		dfs(u, c);
+	}
+}
+
+ll cost(ll a, ll b){
+	if(comps[a].size() > comps[b].size()){
```

Comment:
don't duplicate all of this code

also, it isn't even necessary to swap here; just swap the arguments as necessary on line 218.